### PR TITLE
[PWX-35031]:  T34542894: [add-disk]Check for alerts when pool resize is triggered and done

### PR DESCRIPTION
**Pull request by ragsingh-px**
**What this PR does / why we need it**:
Automation test to Check for alerts when pool resize is triggered and done
**Which issue(s) this PR fixes** (optional)
Closes # PWX-35031

**Special notes for your reviewer**:
Testing Link:- https://jenkins.pwx.dev.purestorage.com/user/ragsingh@purestorage.com/my-views/view/all/job/Users/job/Raghvendra/job/raghav_pxdv-7-dev/58/

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -932,3 +934,127 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
...
+		stepLog = "Move pool to maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			nodeDetail, err = GetNodeWithGivenPoolID(poolToResize.Uuid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Failed to get Node Details using PoolUUID [%v]", poolToResize.Uuid))
+
+			log.InfoD("Bring Node to Maintenance Mode")
+			log.InfoD(fmt.Sprintf("Entering pool maintenance mode on node %s", nodeDetail.Name))
+			err = Inst().V.EnterPoolMaintenance(*nodeDetail)
+			log.FailOnError(err, fmt.Sprintf("fail to enter node %s in maintenance mode", nodeDetail.Name))
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:974]: should test wait for PX to come up successfully in maintenance mode before proceeding?

Comment (vinayakshnd) [reply tests/basic/pool_function_test.go:1036]: Lets use WaitForPoolStatusToUpdate API to wait till PX comes up in maintenance mode.

For eg. as used here: https://github.com/portworx/torpedo/blob/master/tests/common.go#L7564

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -994,3 +995,127 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
...
+			resizeErr := waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(resizeErr, nil, "Pool expansion does not result in error")
+
+		})
+
+		stepLog = "Exit pool out of maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			t := func() (interface{}, bool, error) {
+				status, err := Inst().V.GetNodePoolsStatus(*nodeDetail)
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:1062]: Lets reduce this code by just using WaitForPoolStatusToUpdate as used here: https://github.com/portworx/torpedo/blob/master/tests/common.go#L7587C8-L7587C33

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -994,3 +995,127 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
...
 })
+
+var _ = Describe("{PoolExpandAndCheckAlertsUsingAddDisk}", func() {
+
+	var testrailID = 34542894
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/34542894
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolExpandAndCheckAlertsUsingAddDisk", "pool expansion using add-disk and check alerts after each operation", nil, testrailID)
+		contexts = scheduleApps()
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:1006]: After this command, Lets validate that scheduled Apps come up successfully using ValidateApplications API.
https://github.com/portworx/torpedo/blob/master/tests/common.go#L1891

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go]: fixed.

File: tests/basic/pool_function_test.go (hunk 1 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go]: fixed

File: tests/basic/pool_function_test.go (hunk 3 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:1241]: fixed

**Review by vinayakshnd — APPROVED** [review]

**Pull request merged**